changelog/2603.added.md

```diff
@@ -0,0 +1 @@
+Added the TcpMaxSeg socket option for apple targets
```

Comment:
Actually, this socket option was available for macOS, but [for getsockopt() only](https://docs.rs/nix/latest/x86_64-apple-darwin/nix/sys/socket/sockopt/struct.TcpMaxSeg.html#impl-GetSockOpt-for-TcpMaxSeg). This PR adds the `setsockopt()` part

Reply:
Ah, right you are. Fixed.